ios/RCTTWVideoModule.m

```diff
@@ -305,35 +344,21 @@ -(NSMutableDictionary*)convertLocalVideoTrackStats:(TVILocalVideoTrackStats *)st
 
 RCT_EXPORT_METHOD(getStats) {
     if (self.room) {
-        [self.room getStatsWithBlock:^(NSArray<TVIStatsReport *> * _Nonnull statsReports) {
-            NSMutableDictionary *eventBody = [[NSMutableDictionary alloc] initWithCapacity:10];
-            for (TVIStatsReport *statsReport in statsReports) {
-                NSMutableArray *audioTrackStats = [[NSMutableArray alloc] initWithCapacity:10];
-                NSMutableArray *videoTrackStats = [[NSMutableArray alloc] initWithCapacity:10];
-                NSMutableArray *localAudioTrackStats = [[NSMutableArray alloc] initWithCapacity:10];
-                NSMutableArray *localVideoTrackStats = [[NSMutableArray alloc] initWithCapacity:10];
-                for (TVIRemoteAudioTrackStats *stats in statsReport.remoteAudioTrackStats) {
-                    [audioTrackStats addObject:[self convertRemoteAudioTrackStats:stats]];
-                }
-                for (TVIRemoteVideoTrackStats *stats in statsReport.remoteVideoTrackStats) {
-                    [videoTrackStats addObject:[self convertRemoteVideoTrackStats:stats]];
+        [self.room getStatsWithBlock: self.statsReceivedCallback];
                 }
-                for (TVILocalAudioTrackStats *stats in statsReport.localAudioTrackStats) {
-                    [localAudioTrackStats addObject:[self convertLocalAudioTrackStats:stats]];
-                }
-                for (TVILocalVideoTrackStats *stats in statsReport.localVideoTrackStats) {
-                    [localVideoTrackStats addObject:[self convertLocalVideoTrackStats:stats]];
-                }
-                eventBody[statsReport.peerConnectionId] = @{
-                    @"remoteAudioTrackStats": audioTrackStats,
-                    @"remoteVideoTrackStats": videoTrackStats,
-                    @"localAudioTrackStats": localAudioTrackStats,
-                    @"localVideoTrackStats": localVideoTrackStats
-                };
             }
-            [self sendEventCheckingListenerWithName:statsReceived body:eventBody];
-        }];
+
+RCT_EXPORT_METHOD(requestStats:(int)intervalMs) {
+    [self.getStatsTimer invalidate];
+    self.getStatsTimer = [NSTimer scheduledTimerWithTimeInterval: intervalMs / 1000
```

Comment:
Timer precision might be lost here.
You divide an `int` value by another `int`. Consider `1500 ms, divide by 1000 = 1`, another example `500/1000 = 0`.

Please use NSTimeInterval (aka double) when doing computation.

Reply:
Fixed